lib/configuration/normalizeConfig.js

```diff
@@ -0,0 +1,184 @@
+const R = require('ramda');
+
+/**
+ * Removes options that are no longer supported by Dredd.
+ * Any coercion will not be performed, as they are removed prior to coercion.
+ */
+const removeUnsupportedOptions = R.compose(
+  R.dissoc('q'),
+  R.dissoc('silent'),
+  R.dissoc('t'),
+  R.dissoc('timestamp'),
+  R.dissoc('blueprintPath'),
+  R.dissoc('b'),
+  R.dissoc('sandbox')
+);
+
+const getUserHeader = R.compose(
+  token => `Authorization: Basic ${token}`,
+  user => Buffer.from(user).toString('base64')
+);
+
+const updateHeaderWithUser = R.compose(
+  R.unnest,
+  R.adjust(0, getUserHeader),
+  R.values,
+  R.pick(['user', 'header'])
+);
+
+const coerceToArray = R.cond([
+  [R.is(String), v => [v]],
+  [R.isNil, R.always([])],
+  [R.T, R.identity],
+]);
+
+function coerceToBoolean(value) {
+  if (value === 'true') return true;
+  if (value === 'false') return false;
+  if (value) return true;
+  return false;
+}
+
+/**
+ * Appends authorization header when supplied with "user" option.
+ */
+const coerceUserOption = R.when(
+  R.propSatisfies(R.complement(R.isNil), 'user'),
+  R.compose(
+    R.dissoc('user'),
+    R.over(
+      R.lens(updateHeaderWithUser, R.assoc('header')),
+      R.identity
+    )
+  )
+);
+
+const mapIndexed = R.addIndex(R.map);
+
+const coerceApiDescriptions = R.compose(
+  mapIndexed((content, index) => ({
+    location: `configuration.apiDescriptions[${index}]`,
+    content: R.when(R.has('content'), R.prop('content'), content),
+  })),
+  coerceToArray
+);
+
+/**
+ * Coerces the given deprecated value of the "level" option
+ * and returns the supported value for "loglevel" option.
+ */
+const coerceDeprecatedLevelOption = R.when(
+  R.either(R.has('l'), R.has('level')),
+  R.compose(
+    R.dissoc('l'),
+    R.dissoc('level'),
+    R.over(
+      R.lens(
+        R.compose(
+          R.cond([
+            [
+              R.includes(R.__, ['silly', 'debug', 'verbose']),
+              R.always('debug'),
+            ],
+            [R.equals('error'), R.always('error')],
+            [R.equals('silent'), R.always('silent')],
+            [R.T, R.always('warn')],
+          ]),
+          R.either(R.prop('l'), R.prop('level'))
+        ),
+        R.assoc('loglevel')
+      ),
+      R.identity
+    )
+  )
+);
+
+const coerceDataToApiDescriptions = R.compose(
+  R.unnest,
+  R.values,
+  R.evolve({
+    data: R.compose(
+      R.map(([location, content]) => {
+        const nextDesdcription = (typeof content === 'string')
```

Comment:
typo: nextDesdcription

Reply:
and description is too generic word, should be `nextAPIdescription` or perhaps just `apiDescription` would be enough